modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -660,13 +680,35 @@ public static string GetStringFromASCII(this byte[] bytes)
         }
 
         /// <summary>
-        /// Converts the given byte array of UTF-8 encoded text to a string.
+        /// Converts UTF-16 encoded array to string using the little endian byte order.
+        /// </summary>
+        /// <param name="bytes">A byte array of UTF-16 characters.</param>
+        /// <returns>A string created from the bytes.</returns>
+        public static string GetStringFromUTF16(this byte[] bytes)
+        {
+            return Encoding.Unicode.GetString(bytes);
+        }
+
+        /// <summary>
+        /// Converts UTF-32 encoded array to string using the little endian byte order.
+        /// </summary>
+        /// <param name="bytes">A byte array of UTF-32 characters.</param>
+        /// <returns>A string created from the bytes.</returns>
+        public static string GetStringFromUTF32(this byte[] bytes)
```

Comment:
Since this is an extension method, doing this would force you to convert the byte array to a span before you can use it and of course the method won't show up in IntelliSense for the arrays:

```csharp
byte[] myBytes = { ... };
((ReadOnlySpan<byte>)myBytes).GetStringFromUTF32();
```

Note that using `AsSpan` won't work either because that returns a `Span<byte>` not a `ReadOnlySpan<byte>`.

Reply:
You're right, I forgot this was a `this` parameter.